File: backend/internal/domain/task.go
package domain

import (
	"context"
	"errors"
	"time"
)

type Task struct {
	Id        string       `json:"id"`
	Title     string       `json:"title"`
	Type      TaskCategory `json:"type"`
	DateStart time.Time    `json:"dateStart"`
	DateEnd   time.Time    `json:"dateEnd"`
	Descricao string       `json:"descricao"`
}

type TaskRepository interface {
	CreateTask(ctx context.Context, task *Task) error
	UpdateTask(ctx context.Context, id string, title string, typeid string, dateStart time.Time, dateEnd time.Time, descricao string) error
	Delete(ctx context.Context, id string) error
	ListTask(ctx context.Context) ([]Task, error)
	FindTaskById(ctx context.Context, id string) (*Task, error)
}

type TaskUsecase interface {
	CreateTask(ctx context.Context, task *Task) error
	UpdateTask(ctx context.Context, id string, title string, typeid string, dateStart time.Time, dateEnd time.Time, descricao string) error
	Delete(ctx context.Context, id string) error
	ListTask(ctx context.Context) ([]Task, error)
	FindTaskById(ctx context.Context, id string) (*Task, error)
}

var (
	ERR_TASK_NOT_FOUND = errors.New("task not found ")
)

File: webapp/src/features/Tasks/domain/task.ts
import type { TaskCategory } from "../../tasks_category/domain/task_category";

export interface Task {
  id?: string;
  title: string;
  type: TaskCategory;
  dateStart: Date;
  dateEnd: Date;
  descricao: string;
}

export interface TaskReq {
  title: string;
  type: string;
  dateStart: Date;
  dateEnd: Date;
  descricao: string;
}

export interface ITaskRepository {
  CreateTask(task: TaskReq): Promise<boolean>;
  UpdateTask(id: string, task: TaskReq): Promise<boolean>;
  Delete(id: string): Promise<boolean>;
  ListTask(): Promise<Task[]>;
  FindTaskById(id: string): Promise<Task>;
}

export interface ITaskUsecase {
  CreateTask(task: Task): Promise<boolean>;
  UpdateTask(task: Task): Promise<boolean>;
  Delete(id: string): Promise<boolean>;
  ListTask(): Promise<Task[]>;
  FindTaskById(id: string): Promise<Task | null>;
}
